### PR TITLE
CORE-13560- Stop creating field serializers all the time

### PR DESCRIPTION
**Pull request by conal-smith**
Tested locally and on EKS cluster.

### CODE REVIEW

**Review by JamesHR3 — APPROVED** [review]

**Review by JamesHR3 — APPROVED** [review]

File: quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java [hunk 1]
```diff
@@ -2181,6 +2183,11 @@ public static ByteArraySerializer getFiberSerializer(ClassResolver classResolver
 
     private static final class FiberSerializer extends Serializer<Fiber<?>> {
         private final boolean includeThreadLocals;
+        private final Map<Class<? extends Fiber>, FieldSerializer<Fiber<?>>> fieldSerializers = new HashMap<>();
```
Comment (chrisr3) [inline quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java]: That should be `Class<? extends Fiber<?>>`.

File: quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java [hunk 2]
```diff
@@ -2181,6 +2183,11 @@ public static ByteArraySerializer getFiberSerializer(ClassResolver classResolver
...
+
+        private FieldSerializer<Fiber<?>> getFieldSerializer(Kryo kryo, Class<? extends Fiber> fiberClass) {
```
Comment (chrisr3) [inline quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java]: Similarly `Class<? extends Fiber<?>>`.

File: quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java [hunk 3]
```diff
@@ -2181,6 +2183,11 @@ public static ByteArraySerializer getFiberSerializer(ClassResolver classResolver
...
+            return fieldSerializers.computeIfAbsent(fiberClass, (fc) -> new FieldSerializer<>(kryo, fc));
```
Comment (chrisr3) [inline quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java]: Can you remove the brackets around the lambda `fc` parameter?

**Review by chrisr3 — CHANGES_REQUESTED** [review]
Small things

File: quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java [hunk 4]
```diff
@@ -2202,7 +2209,7 @@ public void write(Kryo kryo, Output output, Fiber<?> f) {
                 try {
                     f.stack.resumeStack();
                     kryo.writeClass(output, f.getClass());
-                    new FieldSerializer<>(kryo, f.getClass()).write(kryo, output, f);
+                    getFieldSerializer(kryo, f.getClass()).write(kryo, output, f);
```
Comment (chrisr3) [inline quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java]: Oh FFS! So getting rid of "raw" types also requires:
```java
(Class<? extends Fiber<?>>)f.getClass()
```
and adding "unchecked" to this function's `@SuppressWarnings`.

File: quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java [hunk 5]
```diff
@@ -90,6 +92,7 @@
  *
  * @author pron
  */
+@SuppressWarnings("unchecked")
```
Comment (chrisr3) [inline quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java]: This is too broad. You only need to replace
```java
@SuppressWarnings("CallToPrintStackTrace")
```
with
```java
@SuppressWarnings({"CallToPrintStackTrace", "unchecked"})
```
on the affected functions.

**Review by chrisr3 — COMMENTED** [review]
Quick change...

**Review by chrisr3 — APPROVED** [review]
LGTM

**Pull request merged**